realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -1209,6 +1211,70 @@ public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
         Realm.migrateRealm(config, migration);
     }
 
+    @Test
+    public void renameAndAddField() {
+        RealmMigration migration = new RealmMigration() {
+            @Override
+            public void migrate(DynamicRealm realm, long oldVersion, long newVersion) {
+                realm.schema.get("MigrationFieldRenameAndAdd")
+                        .renameField("string1", "string2")
+                        .addField("string1", String.class);
```

Comment:
It would be more interesting if it had another type than String. The original `string1` was also a String, so if something goes wrong this probably just works anyway.